File: java/programmers/Runner.java
import java.util.Arrays;

public class Runner {
  public static void main(String[] args) {
    var result = new 보석쇼핑().solution(new String[] {"DIA", "RUBY", "RUBY", "DIA", "DIA", "EMERALD", "SAPPHIRE", "DIA"});
    System.out.println("=======================result");
//    System.out.println(result);
    Arrays.stream(result).forEach(System.out::println);
  }
}

File: java/programmers/Runner.kt
fun main() {
    val edges = arrayOf(
        intArrayOf(0, 1),
        intArrayOf(1, 2),
        intArrayOf(1, 4),
        intArrayOf(0, 8),
        intArrayOf(8, 7),
        intArrayOf(9, 10),
        intArrayOf(9, 11),
        intArrayOf(4, 3),
        intArrayOf(6, 5),
        intArrayOf(4, 6),
        intArrayOf(8, 9)
    )
    val result = 양과늑대().solution(
        intArrayOf(0, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1),
        edges,
    )

    println(result)
}